Library/Homebrew/os/linux/elf.rb

```diff
@@ -0,0 +1,151 @@
+module ELFShim
+  # See: https://en.wikipedia.org/wiki/Executable_and_Linkable_Format#File_header
+  MAGIC_OFFSET = 0
+  MAGIC_SIZE = 4
+  MAGIC_NUMBER = "\x7fELF".freeze
+
+  OSABI_OFFSET = 0x07
+  OSABI_SIZE = 1
+  OSABI_SYSTEM_V = 0
+  OSABI_LINUX = 3
+
+  TYPE_OFFSET = 0x10
+  TYPE_SIZE = 2
```

Comment:
Use `TYPE_EXECUTABLE.to_s.size`

Reply:
These are not equivalent.
`TYPE_EXECUTABLE.to_s.size` ≠ `TYPE_SIZE`
`TYPE_EXECUTABLE.to_s.size` = 1
`TYPE_SIZE` = 2

Reply:
I've added a `read_uint16` helper function.